helpers_test.go

```diff
@@ -0,0 +1,53 @@
+package kv
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	log "github.com/hashicorp/go-hclog"
+	"github.com/hashicorp/vault/sdk/helper/logging"
+	"github.com/hashicorp/vault/sdk/logical"
+)
+
+func getBackend(t *testing.T) (logical.Backend, logical.Storage) {
+	config := &logical.BackendConfig{
+		Logger:      logging.NewVaultLogger(log.Trace),
+		System:      &logical.StaticSystemView{},
+		StorageView: &logical.InmemStorage{},
+		BackendUUID: "test",
+	}
+
+	b, err := VersionedKVFactory(context.Background(), config)
+	if err != nil {
+		t.Fatalf("unable to create backend: %v", err)
+	}
+
+	// Wait for the upgrade to finish
+	time.Sleep(time.Second)
```

Comment:
The `getBackend` method existed before this PR. It is used by the unit tests to instantiate a KV backend. I moved it from one file to another. That said, I agree that it would be good to ensure that the upgrade finishes rather than a single hopeful sleep.

The backend itself has a flag that states that the upgrade is pending. I think it would be cleaner to add polling with a timeout as you said. That's what we do for the `kv` CLI tests in the `vault` repo. I can try to get something like that in place here.

Reply:
I added a ticker/timeout which ticks per second. Each tick, a read request will be submitted to the config endpoint to ensure that it doesn't result in an error.